### PR TITLE
Feature request: Web.HTTP support OAuth bearer token

### PR DESCRIPTION
**Pull request by tsuyoshicho**
In curl

```
--oauth2-bearer <token> OAuth 2 Bearer Token
```

I'm trying to this support.

### CODE REVIEW

**Comment by tyru** [comment]
I didn't know that curl option. >`--oauth2-bearer`
I always specify by -H option.
curl --H "Authorization: Bearer YOUR_ACCESS_TOKEN" 

https://qiita.com/hash/items/ebcdb5135c4a42001f02

maybe just add the above header in other implememtation.

**Comment by tsuyoshicho** [comment]
Okay

I'll implement at future.

**Comment by tsuyoshicho** [comment]
Done and check manually ok.

**Comment by ujihisa** [comment]
ping

**Comment by ujihisa** [comment]
ping

**Comment by mattn** [comment]
LGTM

**Comment by tsuyoshicho** [comment]
Thanks reviewed.

File: doc/vital/Web/HTTP.txt [hunk 1]
```diff
@@ -74,6 +74,9 @@ request({method}, {url} [, {settings}])
 	"password"	Default: (None)
 		Password for an HTTP authentication.
 
+	"token"		Default: (None)
```
Comment (thinca) [inline doc/vital/Web/HTTP.txt]: ただの `token` だと何の token かわかりづらいので `bearerToken` とかがいいのかもしれない。

Comment (tsuyoshicho) [reply doc/vital/Web/HTTP.txt]: Changing OK.

Wait a little.

Comment (tsuyoshicho) [reply doc/vital/Web/HTTP.txt]: fix it

File: doc/vital/Web/HTTP.txt [hunk 2]
```diff
@@ -74,6 +74,9 @@ request({method}, {url} [, {settings}])
 	"password"	Default: (None)
 		Password for an HTTP authentication.
 
+	"bearerToken"		Default: (None)
+		Bearer Token for an HTTP authentication (OAuth2).
```
Comment (thinca) [inline doc/vital/Web/HTTP.txt]:
```suggestion
		Bearer token for an HTTP authentication (OAuth2).
```

Comment (tsuyoshicho) [reply doc/vital/Web/HTTP.txt]: ou
OK

Comment (tsuyoshicho) [reply doc/vital/Web/HTTP.txt]: commited

**Comment by tsuyoshicho** [comment]
All commits rebase to 1 commit.

**Comment by tsuyoshicho** [comment]
いまさらですが、curl以外はauthMethodがあると動かない動作だったので、なんかcurl以外の確認ミスってたと思いますorz

実装はした、というものになってます...

**Pull request merged**